planner/core/prepare_test.go

```diff
@@ -157,6 +157,86 @@ func (s *testPrepareSerialSuite) TestPrepareCacheIndexScan(c *C) {
 	// When executing one statement at the first time, we don't use cache, so we need to execute it at least twice to test the cache.
 	tk.MustQuery("execute stmt1 using @a, @b").Check(testkit.Rows("1 3", "1 3"))
 	tk.MustQuery("execute stmt1 using @a, @b").Check(testkit.Rows("1 3", "1 3"))
+
+	// Prepared plan cache with invisible index
+	cleanHistoryAndEnableStmt := func() {
+		tk.MustExec("set global tidb_enable_stmt_summary = 0") // clean statements history
+		tk.MustExec("set global tidb_enable_stmt_summary = 1")
+	}
+	cleanHistoryAndEnableStmt()
+	defer tk.MustExec("set global tidb_enable_stmt_summary = 0")
+
+	tk.MustExec("drop table t")
+	tk.MustExec("create table t(a int, b int, unique (a), unique (b) invisible)")
+	tk.MustExec("insert into t values(1, 2), (3, 4)")
+
+	tk.MustExec(`prepare stmt2 from "select a from t order by ?"`)
+	tk.MustExec("set @a='a'")
```

Comment:
No, I mean,
1. create a table with visible index
1. execute one SQL, which will use the index
2. make the index invisible
3. execute the SQL again, it shouldn't use the index
4. make the index visible again
5. execute the SQL again, it should use the index

Reply:
BTW, move your code to a new test case.

Reply:
`alter table t alter index idx_a invisible` is not supported in this PR.
I will do it in next PR, how about add these tests in next PR?